### PR TITLE
Add new pr-commenter custom task

### PR DESCRIPTION
**Pull request by abayer**
# Changes

Part of #483

This custom task is meant to be used alongside PR status report updates to add/update/delete a comment equivalent to the Prow one listing job failures with links to logs and the command to re-test.

I initially made an attempt to update the `github-add-comment` task in the catalog to do this (https://github.com/tektoncd/catalog/pull/1051) but decided that a custom task frankly made a lot more sense. Its behavior is a pretty direct emulation of how Prow and Lighthouse do this, looking for existing comments by the bot user with a given tag in the comment, and updating/deleting that comment if it exists, creating a new comment otherwise. It's a bit simplified, and instead of using `ProwJob`s to know what should still be in the comment and what should be removed, it specifically just adds or removes the specific job it's reporting on while preserving whatever else was there.

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by abayer** [comment]
Ok, turns out getting an integration test set up properly is more hassle than it's worth, since this exact same behavior with extremely similar code is covered by Lighthouse's end-to-end tests, so I'm just going to do a manual test locally and then get rid of the WIP.

**Comment by abayer** [comment]
> Shouldn't there be a CRD that defines `PRCommenter` ?

Originally there was, but then I realized that was unnecessary since it's just parameters, but I kept the struct in the same form anyway. 🤷

**Comment by abayer** [comment]
^^ is me testing the bot.

File: tekton/resources/nightly-release/overlays/pr-commenter/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namePrefix: pr-commenter-
+bases:
+  - ../../base
+patchesJson6902:
+  - target:
+      group: triggers.tekton.dev
+      version: v1alpha1
```
Comment (afrittoli) [inline tekton/resources/nightly-release/overlays/pr-commenter/kustomization.yaml:7]: Heh, I need to switch the bases to beta eventually

**Review by afrittoli — COMMENTED** [review]
This looks great, thank you!
I would love to see a README with an example `Run`, but otherwise it seems ok, including the release resources and cronjob.
/approve

**Comment by abayer** [comment]
> I would love to see a README with an example `Run`, but otherwise it seems ok, including the release resources and cronjob.

I knew I missed something! =) Adding that now.

**Review by afrittoli — COMMENTED** [review]
Let's get the ball rolling!
Hopefully we can get a nightly and deploy to dogfooding!
/lgtm

**Pull request merged**